src/cachingreader.cpp

```diff
@@ -301,7 +301,7 @@ void CachingReader::process() {
     }
 }
 
-int CachingReader::read(int sample, int num_samples, CSAMPLE* buffer) {
+int CachingReader::read(int sample, bool reverse, int num_samples, CSAMPLE* buffer) {
```

Comment:
I don't see the reason for making CachingReader aware of reverse reads -- that seems like something that is playback specific and shouldn't need to be something the underlying data reader is aware of.